src/main/java/com/rabbitmq/jms/client/RMQConnection.java

```diff
@@ -82,6 +84,13 @@
     private volatile boolean canSetClientID = true;
 
     /**
+     * QoS setting for channels created by this connection.
+     *
+     * @see com.rabbitmq.client.Channel#basicQos(int)
+     */
+    private final int channelsQos;
```

Comment:
It doesn't have to be `final`. The intent is to have a `RMQConnection` creating AMQP channels with the same settings, not more. Not sure users are coding against the `RMQConnection` class, but more against the JMS `*Connection` interfaces. We can make it `volatile` and expose it with getter/setter, it should be harmless.